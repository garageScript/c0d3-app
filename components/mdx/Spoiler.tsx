import React, { useState } from 'react'
import styles from '../../scss/spoiler.module.scss'
type Props = {
  name?: string
}
const Spoiler: React.FC<Props> = ({ children, name }) => {
  const [status, setStatus] = useState({
    css: { display: 'none' },
    name: 'spoiler-arrow__right'
  })
  const onClick = () => {
    status.css.display === 'none'
      ? setStatus({ css: { display: 'inherit' }, name: 'spoiler-arrow__down' })
      : setStatus({ css: { display: 'none' }, name: 'spoiler-arrow__right' })
  }
  return (
    <>
      <div className="mb-1 ml-1" onClick={onClick}>
        <div className="d-flex">
          <div className={`${styles[status.name]} align-self-center mr-3`} />
          <div className={`${styles['spoiler__title']}`}>
            {name || 'Answer'}
          </div>
        </div>
      </div>
      <div style={status.css}>{children}</div>
    </>
  )
}

export default Spoiler
