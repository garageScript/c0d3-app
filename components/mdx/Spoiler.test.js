import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import Spoiler from './Spoiler'

describe('Spoiler component test', () => {
  test('Should render spoiler', async () => {
    const { container } = render(
      <Spoiler name="TestSpoiler">
        <h1>Hidden text</h1>
      </Spoiler>
    )
    expect(container).toMatchSnapshot()
    userEvent.click(screen.getByText('TestSpoiler'))
    expect(screen.getByText('Hidden text')).toBeVisible()
    userEvent.click(screen.getByText('TestSpoiler'))
  })
  test('Should render spoiler without name', () => {
    render(
      <Spoiler>
        <h1>Hidden text</h1>
      </Spoiler>
    )
    expect(screen.getByText('Answer')).toBeTruthy()
  })
})
