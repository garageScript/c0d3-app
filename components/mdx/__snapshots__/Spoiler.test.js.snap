// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spoiler component test Should render spoiler 1`] = `
<div>
  <div
    class="mb-1 ml-1"
  >
    <div
      class="d-flex"
    >
      <div
        class="spoiler-arrow__right align-self-center mr-3"
      />
      <div
        class="spoiler__title"
      >
        TestSpoiler
      </div>
    </div>
  </div>
  <div
    style="display: none;"
  >
    <h1>
      Hidden text
    </h1>
  </div>
</div>
`;
